### ISSUE TITLE
Keyboard shortcuts?

### ISSUE THREAD

**Issue by DaveWalk1** [standardnotes/forum#39]
Is there a plan for keyboard shortcuts? Some off the top of my head:
- Search, as in put keyboard focus on search bar. (`CTRL + F`)
- New note (`CTRL + N`)
- Toggle Markdown preview window (`CTRL + SHIFT + M`)

And some of the more exotic ones I remember from text editors like Atom:
- move line up/down (`CTRL + up/down`)
- highlighting text and doing the same

**Comment by DaveWalk1** [comment]
- put focus on tag text section (`CTRL + SHIFT + T`)

**Comment by mobitar** [comment]
I'm still thinking of a way to do this through extensions but have not yet arrived at a full solution. But it is something I hope to get to.

**Comment by andreasio** [comment]
This is really important for a good experience. And it does not interfer with the text format. I'd also add shortcut to bring out the menu, and shortcut to delete a note.

**Comment by sickill** [comment]
I love all ideas behind Standard Notes (focus on simplicity, privacy, e2e encryption, longevity, standardfile), and I think it's the best option out there (Laverna is super buggy, Turtl's usability is so-so), but... lack of keyboard shortcuts on desktop/web is a deal breaker for me. I'm switching back and forth between SN and nvalt (which I can operate exclusively with the keyboard), always going back to nvalt for this single reason. Lack of any keyboard shortcuts whatsoever is also major reason why I still didn't get SN Extended subscription 😐 

Wanted to give my honest feedback. Regardless, keep up the good work Mo!

**Comment by mobitar** [comment]
Thanks @sickill, really appreciate it! Keyboard shortcuts have definitely not been forgotten about. I had a beta of it up but postponed it for various reasons. Will probably re-introduce it at some point this year.

**Comment by ebelinski** [comment]
I agree with @andreasio and @sickill. I didn't realize how important keyboard shortcuts were to me until I realized they were missing in Standard Notes. I have been trying out Standard Notes for over a year, but I can't get myself to seriously use it because missing keyboard shortcuts is a productivity killer for me.

**Comment by mobitar** [comment]
@ebelinski what kind of shortcuts would be essential to your workflow? It's not out of the question, but would most likely come in the form of an extension. (p.s thanks for the shoutout in your privacy article!)

**Comment by kooshiel** [comment]
I'd also love to see an extension that would allow for keyboard shortcuts. In addition to those mentioned above. I'd be happy to se:

- `⌘L` to set focus to search bar and highlight search term (if one has already been entered)
- `ESC` to clear search bar

**Comment by nikolayhg** [comment]
* `F2` - rename a note

**Comment by sickill** [comment]
I agree with @ebelinski re ability to customize shortcuts. I believe it's more important to have a decent amount of useful actions exposed to be key-mapped, than to find a perfect keys for them right from the start (I'm sure there are many people like me who would re-map it all to the keys they're used to use in their current note taking apps).

So, regarding useful actions: I really like how nvALT app handles search and creating of new note from a single input box. You type something, and you get results live as you type. So, if it finds the text in any of the notes then it filters the list to include only these, and it opens the first one. If it doesn't find the text in any of the notes, then I can just hit <enter> to create a note with the search text as its title. I find this super productive.

**Comment by svnsbck** [comment]
+1 Keyboard shortcuts I think are crucial to the desktop app (even for the mobile ones). The more the better. Could also be as an extension.

**Comment by hwcltjn** [comment]
Any updates on this? Just purchased extended and realised there were no keyboard shortcuts!

**Comment by mobitar** [comment]
No updates to give yet..

**Comment by virtuallyvlad** [comment]
I agree on customizing shortcuts, but if that's not possible then there should at least be a screen to show all available shortcuts.  Having to guess which shortcuts are available or digging through documentation would be less ideal.

**Comment by williamjphamilton** [comment]
+1 Customisable keyboard shortcuts please! 

Critical feature for productivity.

**Comment by mobitar** [comment]
@williamjphamilton what shortcuts would you be looking for?

**Comment by mplorentz** [comment]
This feature is keeping me from switching to Standard Notes as well. I think Notational Velocity has a great set of shortcuts. ⌘L to focus in the search bar, arrow keys to navigate search results, and Tab to move the cursor to the editor. 

I'd love to see the same shortcuts supported in the iOS app as well, as I frequently use my iPad with an external keyboard.

**Comment by williamjphamilton** [comment]
> @williamjphamilton what shortcuts would you be looking for?

I would like to be able 'alt-tab' through notes and folder + and agree with what another user suggested: 

> * `⌘N` for a new note
> * `⌘F` to search within a note
> * `⌥⌘F` to search multiple notes
> * `Delete` to delete a selected note
> * Up and down arrow keys to navigate between selected notes
> * `⌘,` to open the settings panel
> 
> Ideally these options would also show up in the macOS menu bar. This would make it possible for users to set their own custom keyboard shortcuts using System Preferences (in Keyboard > Shortcuts > App Shortcuts).

**Comment by mobitar** [comment]
Thanks @williamjphamilton and @mplorentz. This is definitely planned, so still collecting feedback on what kind of shortcuts should be possible. It will be customizable though.

**Comment by williamjphamilton** [comment]
> Thanks @williamjphamilton and @mplorentz. This is definitely planned, so still collecting feedback on what kind of shortcuts should be possible. It will be customizable though.

Thank you! Looking forward to the feature. Keep up the good work.

**Comment by stefanobernardi** [comment]
Also the only thing that's preventing me from becoming a paid supporter

**Comment by joshstrange** [comment]
This is the only thing keeping me from seriously trying this is the ability to hide/show the window with a global hotkey like I do in NvAlt. I use my global hotkey 10's of times a day easy.

**Comment by levjoy** [comment]
Another vote for keyboard shortcuts. I just became a paid user and SN is definitely the best app for my needs by far. However, the lack of shortcuts is a huge gap! Like others in this thread the keyboard is central to my workflow.  

Customizable shortcuts would be great, though I'd be looking for the same set described by @williamjphamilton. 

I love SN, keep up the amazing work!

**Comment by mobitar** [comment]
Thanks @levjoy, noted!

**Comment by joeySeal** [comment]
+1 for this feature

**Comment by gaganpreet** [comment]
Another vote for keyboard shortcuts. An app focused on taking notes and working with text can't not have enough of these.

**Comment by mark6400** [comment]
Another +1. Remaining at the keyboard is critical in writing and working with notes. Notational Velocity got it right with respect to the command-L feature (search, or create new note with query title if query returns nothing). Without this feature, I consider Standard Notes to be a great concept but an impractical app, and I'll stick with NValt until it appears (via extension or core capability).

**Comment by fletom** [comment]
+1, I personally can't use any app efficiently without keyboard shortcuts

**Comment by mobitar** [comment]
On the roadmap for 2019 :)

**Comment by david-coneff** [comment]
To add to the keyboard shortcut wishlist:
* hide/unhide the Note list & hashtag UI panels

would save a lot of screen real-estate for the editor when used on a portrait oriented monitor. Looks like we can only resize these UI panels with a mouse via finding the resizer element that exists in a very narrow horizontal space -> irritating to have to mouse with that precision and then deal have a slow graphical operation create some lag in the process of hiding the panel.

**Comment by iedd** [comment]
+1 for shortcuts. Really the only important one for me is "Search", But "New Note" would be useful too.
Keep up the great work!

**Comment by izznogooood** [comment]
The one I miss the most is duplicate line (or selected paragraph)! +1

**Comment by nikess** [comment]
Came across SN. Love it. Zero keyboard shortcuts.  Uninstall. A pity!

**Comment by advait** [comment]
@mobitar Can I help contribute to make this happen? Happy to implement a solution and/or participate in design discussions.

**Comment by mobitar** [comment]
@advait thanks for offering to help. The implementation will be pretty straightforward. It's just a matter of working our way towards it.

**Comment by MoritzLauten** [comment]
I just installed SN, but missing shortcuts. I hope this will be avaible soon - then i buy the extended

**Comment by chmac** [comment]
I *love* the concept of Standard Notes. Making me go to the mouse for every action feels like a deal breaker.

> @advait thanks for offering to help. The implementation will be pretty straightforward. It's just a matter of working our way towards it.

@mobitar This is a real tricky reply.  It's been on the roadmap for a year, no progress that I can see. Turning away folks from helping feels like the opposite of progress on this. Unless, reading between the lines, what you're really saying is you expect it to be released in the next month or so? :-)

**Comment by mobitar** [comment]
Sorry, I made a mistake mentioning it was on the roadmap. Should know better by now to make feature promises. Keyboard shortcuts will come when SN is ready for them. That's all I can say.

**Comment by ErinPo** [comment]
...two years of talking (apparently starting here https://github.com/standardnotes/desktop/issues/16) about a basic feature which is standard in native Mac apps? If it really becomes available only as a paid extension after waiting for years, then I think it will be the first app on the Mac to charge extra for basic keyboard shortcuts :((

**Comment by effieeee** [comment]
Tracking here: https://github.com/standardnotes/forum/issues/1511

**Comment by atmoio** [comment]
Keyboard shortcuts are now available: https://github.com/standardnotes/forum/issues/1511#issuecomment-1322276205
